pandas/core/frame.py

```diff
@@ -613,8 +613,52 @@ def __init__(
                     values, index, columns, dtype=values.dtype, copy=False
                 )
 
+        manager = get_option("mode.data_manager")
+
+        if manager == "array" and not isinstance(mgr, ArrayManager):
+            # TODO proper initialization
+            df = DataFrame(mgr)
+            mgr = df._as_manager("array")._mgr
+        # TODO check for case of manager="block" but mgr is ArrayManager
+
         NDFrame.__init__(self, mgr)
 
+    def _as_manager(self, typ):
+        """
+        Private helper function to create a DataFrame with specific manager.
+
+        Parameters
+        ----------
+        mgr : {"block", "array"}
+
+        Returns
+        -------
+        DataFrame
+            New DataFrame using specified manager type. Is not guaranteed
+            to be a copy or not.
+        """
+        new_mgr: Union[BlockManager, ArrayManager]
+        mgr = self._mgr
+        if typ == "block":
+            if isinstance(mgr, BlockManager):
+                new_mgr = mgr
+            else:
+                new_mgr = arrays_to_mgr(
+                    mgr.arrays, mgr.axes[0], mgr.axes[1], mgr.axes[0], dtype=None
+                )
+        elif typ == "array":
+            if isinstance(mgr, ArrayManager):
+                new_mgr = mgr
+            else:
+                arrays = [arr.copy() for arr in self._iter_column_arrays()]
```

Comment:
The `copy` will actually ensure it is converted to C order (copy has a keyword for this, and the default is `order="C"`), and that it becomes a contiguous 1d array, instead of a strided view in the 2D array of the block.

Reply:
is doing this no-copy not viable?

Reply:
right I think we actually need to copy these to coniguous 1d, but i would make this explict.

Reply:
Indeed, generally a no-copy is not possible if you want a proper 1D array layout. 
Of course, this could be optimized (eg not copying for non-consolidated blocks, etc). But at the moment this function is mainly for testing purposes, and the DataFrame construction is something that still needs to be optimized in general, which can be done in a later PR.